sympy/geometry/ellipse.py

```diff
@@ -305,6 +305,43 @@ def center(self):
         """
         return self.args[0]
 
+    def ellipse_circumference(self):
```

Comment:
as for Ellipse.normal_lines, let's add a `prec` keyword to circumference and just put this code there as a special case when `all(i.is_number for i in (self.vradius, self.hradius))`.

Reply:
@smichr  Can you clarify what you are trying to say here again please? (I couldn't understand you comment completely)

Reply:
Don't make a new method. Add a parameter `prec` to `circumference` and have it handled the same way that it is handled in the `normal_lines` routine to calculate the numerical approximation of normal lines (but in this case you will use it to calculate an approximation of the circumference). Feel free to ask again if this isn't clear.

```python
def circumference(self, prec=None):
    if prec is None or not all(i.is_number for i in (self.vradius, self.hradius)):
        return normal symbolic circumference
    else:
        return calculated value to given prec
```

Reply:
@smichr  I don't think that'll be a good idea (not making a new method) because of these reasons:
1. `circumference` is a `property`, while this method brings out the numerical value of the circumference

(I tried that and it resulted in the following error :
```
avi@avi-Aspire-A515-51G:~/sympy$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Point, Ellipse
>>> p1 = Point(0, 0)
>>> e1 = Ellipse(p1, 3, 1)
>>> e.circumference(prec=20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Mul' object is not callable
```)

2. `circumference`  produces an `ellipse object` whereas this method gives the numerical value.
If we still forcibly convert that ( `circumference` ) to a `non-property`, it'll be done by removing `@property` statement from the beginning. Then it would result in an `ellipse object` when the command is (I did that too and this is how we would get the results :
```
avi@avi-Aspire-A515-51G:~/sympy$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Point, Ellipse
>>> p1 = Point(0, 0)
>>> e1 = Ellipse(p1, 3, 1)
>>> from sympy import Point, Ellipse, N
>>> N(e1.circumference())
13.3648932205553
```)

Now this could be acceptable but then quite some tests would have to be changed. (I have no issues doing either one)


So what do you think I should do?

Reply:
Hmm... `normal_lines` is a function so there is some precedent for Ellipse's related objects to be functions. 

The more helpful (SymPy-wide) thing to do would be to apply this method to the evaluation of the corresponding elliptic function when conditions are right (and I don't recall under what conditions this approximation succeeds). 

An alternative would be to make a private method (not property) named `_circumference` which can silently go away once the evaluation of ellipticals is fixed. 

Or name your method `circumference_n`...the "ellipse" part is redundant.

At this [summary](http://www.numericana.com/answer/ellipse.htm) page there is an exact representation in terms of the Legendre polynomial. Does that have better evaluation characteristics? Does the elliptical know how to rewrite itself into a Legendre polynomial?

Reply:
> has been implemented in `mpmath`

Something must be different or else the issue #6952 has been fixed. There was a *very large* difference in times taken to compute the result...but that was 5 years ago.  @avishrivastava11 , can you see if there is much difference in the times for these two today?

```python
>>> from time import time
>>> t = time();Ellipse((0,0),3,1).circumference.n(), time()-t
>>> t = time();Ellipse((0,0),3,1).ellipse_circumference(), time()-t
```